apps/music/js/metadata/ogg.js

```diff
@@ -182,16 +257,70 @@ var OggMetadata = (function() {
    */
   function readComment(page) {
     if (page.remaining() < 4) { // 4 bytes for comment-length variable
-      // TODO: handle metadata that uses multiple pages
-      throw new EndOfPageError();
+      throw new EndOfPageError({
+        page: page,
+        index: page.tell(),
+        size: page.remaining()
+      });
     }
     var comment_length = page.readUnsignedInt(true);
     if (comment_length > page.remaining()) {
-      // TODO: handle metadata that uses multiple pages
-      throw new EndOfPageError();
+      throw new EndOfPageError({
+        page: page,
+        index: page.tell(),
+        size: page.remaining()
+      }, comment_length);
     }
 
     var comment = page.readUTF8Text(comment_length);
+    return decodeComment(comment);
+  }
+
+  function readContinuedComment(leftovers, comment_length, page) {
```

Comment:
I wonder if we could merge this with `readComment` above, or add a wrapper around the two. That would simplify how we call this code, since there'd only be a single entry point. I know this is pretty similar to how it is now, but it might make it easier to understand in the future.